app/vmagent/remotewrite/remotewrite.go

```diff
@@ -341,6 +349,7 @@ var configReloaderWG sync.WaitGroup
 //
 // It is expected that nobody calls TryPush during and after the call to this func.
 func Stop() {
+	close(ingestionRateLimiterStopCh)
```

Comment:
looks like it can panic if ingestionRateLimiterStopCh == nil (default case for not defined limit.

Reply:
fixed

Reply:
I was actually thinking of `var ingestionRateLimiterStopCh = make(chan struct{})` It must simplify code.